### ISSUE TITLE
Other custom rules do not work properly export for those are in `@intlify/vue-i18n/recommended`

### ISSUE THREAD

**Issue by daenamkim** [intlify/eslint-plugin-vue-i18n#478]
**Tell us about your environment**

- **ESLint version:** `v8.57.0`
- **eslint-plugin-vue version:** `"eslint-plugin-vue": "^9.17.0"`
- **eslint-plugin-vue-i18n version:** `"@intlify/eslint-plugin-vue-i18n": "^2.0.0"`
- **Node version:** `v20.11.1`

**Please show your full configuration:**

```json5
/* eslint-env node */
module.exports = {
  root: true,
  'extends': [
    'plugin:vue/vue3-essential',
    'eslint:recommended',
    'plugin:@intlify/vue-i18n/recommended'
  ],
  parserOptions: {
    ecmaVersion: 'latest'
  },
  settings: {
    'vue-i18n': {
      localeDir: './src/assets/language/*.{json}',
    },
  },
  rules: {
    // Following rules are not working!
    '@intlify/vue-i18n/no-unused-keys': [
      'error',
      {
        extensions: ['.ts', '.vue'],
      },
    ],
    '@intlify/vue-i18n/key-format-style': [
      'error',
      'camelCase',
      {
        allowArray: false,
        splitByDots: false,
      },
    ],
  }
}

```

**What did you do?**

I just followed https://eslint-plugin-vue-i18n.intlify.dev/started.html and spent few days to figure out why. Tried parser option, different vite plugin for i18n, all did not work.

**What did you expect to happen?**

`@intlify/vue-i18n/no-unused-keys` and `@intlify/vue-i18n/key-format-style` should show errors

```vue
<script setup>
import { useI18n } from 'vue-i18n';

const { t } = useI18n();
</script>

<template>
  <main>
    <!-- ================================ TEST ================================ -->
    <!-- OK -->
    <div>RAW TEXT</div>
    <!-- DOES NOT WORK -->
    <div>{{ t('unknown') }}</div>
    <!-- DOES NOT WORK -->
    <div>{{ t('kebab-case') }}</div>
    <!-- DOES NOT WORK -->
    <div>{{ t('snake_case') }}</div>
    <!-- ================================ TEST ================================ -->
  </main>
</template>

<style scoped>
.logo {
  display: block;
  margin: 0 auto 2rem;
}
</style>

```

**What actually happened?**

It ONLY shows the raw text warning which is in `@intlify/vue-i18n/recommended` but not other rules, and also custom `"@intlify/vue-i18n/no-raw-text": "error"` worked as well when I added in `rules: { ... }`.

```sh
$ yarn lint                                                                                          master
yarn run v1.22.21
$ eslint . --ext .vue,.js,.jsx,.cjs,.mjs --fix --ignore-path .gitignore

/.../src/App.vue
  11:10  warning  raw text 'RAW TEXT' is used  @intlify/vue-i18n/no-raw-text

✖ 1 problem (0 errors, 1 warning)

✨  Done in 0.79s.
```

**Repository to reproduce this issue**

https://github.com/daenamkim/eslint-plugin-vue-i18n-test

**Comment by daenamkim** [comment]
Is this related to Vue3? Since we have used this in Vue2 env it worked as expected.

**Comment by daenamkim** [comment]
@ota-meshi  Hi, I understand you might be very busy, but I would be grateful if you could take a look at this when you're available. Thank you! 🙇

**Comment by daenamkim** [comment]
Hi @ota-meshi,

Thank you for updating this plugin last week!

I’ve upgraded `@intlify/eslint-plugin-vue-i18n` to `3.1.0` and tested most of the best practice rules we would need. However, I’m encountering an issue and was wondering if you could guide me on how to resolve it. 🙇 🙇 🙇 🙇  I’m not sure if this is specific to my setup, as I’ve tried everything I could, including using a [custom parser](https://eslint-plugin-vue-i18n.intlify.dev/started#how-to-use-custom-parser).

### ✅  [@intlify/vue-i18n/no-missing-keys](https://eslint-plugin-vue-i18n.intlify.dev/rules/no-missing-keys.html#intlify-vue-i18n-no-missing-keys) (in recommended)

After I removed `corsPage.corsOrigin` from `en.json`

```sh
/project/web/src/views/CorsPage.vue
  29:17  warning  'corsPage.corsOrigin' does not exist in localization message resources  @intlify/vue-i18n/no-missing-keys
```

### ⚠️ [@intlify/vue-i18n/no-raw-text](https://eslint-plugin-vue-i18n.intlify.dev/rules/no-raw-text.html) (in recommended)
```sh
/projects/web/src/views/CorsPage.vue
  19:10  warning  raw text 'Raw Text 1' is used  @intlify/vue-i18n/no-raw-text
```

- ✅  <img width="260" alt="image" src="https://github.com/user-attachments/assets/08c6ee3b-c9b2-4cc2-9258-db35468cdd76">
- ❌ <img width="442" alt="image" src="https://github.com/user-attachments/assets/6823d8e8-e376-4d23-a0b0-02918e758af0">
- ❌ <img width="329" alt="image" src="https://github.com/user-attachments/assets/4ee02ae7-41eb-4050-b6d5-c8832a09643d">

### ❌ [@intlify/vue-i18n/key-format-style](https://eslint-plugin-vue-i18n.intlify.dev/rules/key-format-style.html#intlify-vue-i18n-key-format-style)
Default is `camelCase` and it seems still not working as expected

- ❌ <img width="400" alt="image" src="https://github.com/user-attachments/assets/75e10119-4511-4ebe-999f-2e694cbe9495">

- ❌ <img width="411" alt="image" src="https://github.com/user-attachments/assets/a9a173e1-cd47-49b3-93b8-dd2e25bb9206">

### ✅  [@intlify/vue-i18n/no-duplicate-keys-in-locale](https://eslint-plugin-vue-i18n.intlify.dev/rules/no-duplicate-keys-in-locale.html)

Even though it is not added to `.eslintrc.cjs`, it still works as intended

```sh
/projects/web/src/assets/language/en.json
  507:6  error  Duplicate object key  json/*
  508:6  error  Duplicate object key  json/*
```

### ❌ [@intlify/vue-i18n/no-missing-keys-in-other-locales](https://eslint-plugin-vue-i18n.intlify.dev/rules/no-missing-keys-in-other-locales.html#intlify-vue-i18n-no-missing-keys-in-other-locales)
`ja.son` is empty and it seems still not working as expected
<img width="226" alt="image" src="https://github.com/user-attachments/assets/d483bd91-bd3e-4b9f-a48c-5d7bf016d029">

### ❌ [@intlify/vue-i18n/no-unused-keys](https://eslint-plugin-vue-i18n.intlify.dev/rules/no-unused-keys.html)
Unused key was not decteded

<img width="377" alt="image" src="https://github.com/user-attachments/assets/a67af4c5-15bc-4dda-aa23-28c18fbcbf63">

### eslintrc.cjs

```cjs
/* eslint-env node */
require('@rushstack/eslint-patch/modern-module-resolution');

module.exports = {
  root: true,
  env: {
    es2020: true,
  },
  extends: [
    'eslint:recommended',
    'plugin:vue/vue3-essential',
    'plugin:vue/vue3-recommended',
    'plugin:vuetify/recommended',
    '@vue/eslint-config-typescript/recommended',
    '@vue/eslint-config-prettier',
    'airbnb-base',
    'plugin:json/recommended',
    'prettier',
    'plugin:storybook/recommended',
    'plugin:playwright/recommended',
    '.eslintrc-auto-import.json',
    'plugin:@intlify/vue-i18n/recommended',
  ],
  parserOptions: {
    // To avoid "Parsing error: Expected to be an expression, but got empty" on .json files
    // https://eslint-plugin-vue-i18n.intlify.dev/started.html#how-to-use-custom-parser
    // https://eslint.org/docs/latest/use/configure/parser#configure-a-custom-parser
    parser: '@typescript-eslint/parser',
    ecmaVersion: 'latest',
  },
  plugins: ['vue'],
  settings: {
    // https://eslint-plugin-vue-i18n.intlify.dev/started.html#settings-vue-i18n
    'vue-i18n': {
      localeDir: './src/assets/language/*.json',
    },
    'import/resolver': {
      typescript: {
        project: './tsconfig.json',
      },
    },
  },
  rules: {
    // TODO: Fix optional rules don't work
    // https://github.com/curvegrid/multibaas/issues/6060
    '@intlify/vue-i18n/no-unused-keys': [
      'error',
      {
        extensions: ['.ts', '.vue'],
      },
    ],
    '@intlify/vue-i18n/key-format-style': [
      'error',
      'camelCase',
      {
        allowArray: false,
        splitByDots: false,
      },
    ],
    'playwright/expect-expect': [
      'error',
      {
        assertFunctionNames: [
          'checkGroup',
          'checkPaginationMenu',
          'checkPagination',
          'checkTxDetails',
          'checkUsers',
          'revokeApiKey',
          'expectPlanUsageBoxes',
          'selectSigner',
          'openSignerSelectorAndSelect',
          'checkInstructionsGuide',
          'transferValueAndCheck',
          'callFunctionAndCheck',
          'checkSidePanelSelectedMenu',
          'checkWeiConverterUnitButton',
          'checkWeiValueSent',
          'callAndConfirmFunction',
          'expectClipboardCopy',
          'checkPopupAndSelectSigner',
          'expectTooltip',
          'checkSendValue',
          'checkCallFunction',
        ],
      },
    ],
    'playwright/prefer-strict-equal': 'error',
    'playwright/prefer-to-be': 'error',
    'playwright/prefer-to-have-count': 'error',
    'playwright/prefer-to-have-length': 'error',
    'playwright/prefer-hooks-in-order': 'error',
    'playwright/prefer-hooks-on-top': 'error',
    'playwright/prefer-equality-matcher': 'error',
    'playwright/prefer-comparison-matcher': 'error',
    'playwright/prefer-lowercase-title': 'error',
    'playwright/prefer-web-first-assertions': 'error',
    'playwright/prefer-locator': 'error',
    'playwright/prefer-native-locators': 'error',
    // HINT: see BaseModal comment
    'vue/require-explicit-emits': 'off',
    'vue/no-useless-v-bind': 'error',
    'vue/no-required-prop-with-default': 'error',
    'import/extensions': 'off',
    'import/consistent-type-specifier-style': 'error',
    'import/order': [
      'error',
      {
        groups: ['type'],
        'newlines-between': 'always',
      },
    ],
    'func-call-spacing': 'off',
    'no-bitwise': [
      'error',
      {
        allow: ['|'],
      },
    ],
    'no-console': [
      'error',
      {
        allow: ['warn', 'error'],
      },
    ],
    'import/no-extraneous-dependencies': [
      'error',
      {
        devDependencies: true,
      },
    ],
    'no-await-in-loop': 'off',
    'no-restricted-syntax': [
      'error',
      {
        selector: 'ForInStatement',
        message:
          'for..in loops iterate over the entire prototype chain, which is virtually never what you want. Use Object.{keys,values,entries}, and iterate over the resulting array.',
      },
      {
        selector: 'LabeledStatement',
        message:
          'Labels are a form of GOTO; using them makes code confusing and hard to maintain and understand.',
      },
      {
        selector: 'WithStatement',
        message:
          '`with` is disallowed in strict mode because it makes code impossible to predict and optimize.',
      },
      {
        selector:
          '*[name=/(?!MultiBaas)(?!Multibaas)(?!multibaas)(?!MULTIBAAS)([M|m][U|u][L|l][T|t][I|i]?.[B|b][A|a][A|a][S|s])/]',
        message:
          "Please use the format 'MultiBaas', 'Multibaas', 'multibaas', or 'MULTIBAAS' in identifiers",
      },
      {
        selector:
          '*[value=/(?!MultiBaas)(?!Multibaas)(?!multibaas)(?!MULTIBAAS)([M|m][U|u][L|l][T|t][I|i]?.[B|b][A|a][A|a][S|s])/]',
        message:
          "Please use the format 'MultiBaas', 'Multibaas', 'multibaas', or 'MULTIBAAS' in literals",
      },
    ],
  },
};
```

**Comment by dahu33** [comment]
@ota-meshi & @kazupon any idea why the linters above are not working as expected?

**Comment by valerijmedvid** [comment]
Any update?

**Comment by daenamkim** [comment]
I found the rules I mentioned above worked for me after upgrading to eslint 9. I have no idea why it did not work in eslint 8 before and it worked in eslint 9 now. It is the mysterious world of the frontend for me. I think it is time to let eslint 8 go. ✋ 

```sh
  7:5  error  'addressesPage.nice' does not exist in 'ja' locale(s)      @intlify/vue-i18n/no-missing-keys-in-other-locales
  7:5  error  unused 'addressesPage.nice' key                            @intlify/vue-i18n/no-unused-keys
  8:5  error  "nice-day" is not camelCase                                @intlify/vue-i18n/key-format-style
  8:5  error  'addressesPage.nice-day' does not exist in 'ja' locale(s)  @intlify/vue-i18n/no-missing-keys-in-other-locales
  8:5  error  unused 'addressesPage.nice-day' key                        @intlify/vue-i18n/no-unused-keys
```

If anyone has the same issue, please try to upgrade to eslint 9. But, be careful, it could introduce new problems! 😆 

Anyway, I will close this myself.
